DLCS.Repository/Assets/DapperAssetRepository.cs

```diff
@@ -52,7 +49,54 @@ public async Task<ImageLocation> GetImageLocation(AssetId assetId)
             return await connection.QuerySingleOrDefaultAsync<ImageLocation>(ImageLocationSql,
                 new { Id = assetId.ToString() });
         }
+        
+        private async Task<Asset> GetAssetInternal(string id)
+        {
+            var key = $"asset:{id}";
+            return await appCache.GetOrAddAsync(key, async entry =>
+            {
+                logger.LogInformation("Refreshing assetCache from database {Asset}", id);
+                await using var connection = await DatabaseConnectionManager.GetOpenNpgSqlConnection(configuration);
+                dynamic? rawAsset = await connection.QuerySingleOrDefaultAsync(AssetSql, new { Id = id });
+                if (rawAsset == null)
+                {
+                    entry.AbsoluteExpirationRelativeToNow =
+                        TimeSpan.FromSeconds(cacheSettings.GetTtl(CacheDuration.Short));
+                    return NullAsset;
+                }
 
+                return new Asset
```

Comment:
We'll need a bit of a tidy up of the various representations of this DB row that are present in different bits of the code!